File: gateway/src/main/java/com/skm/sftp/gateway/GatewayHook.java
package com.skm.sftp.gateway;

import com.inductiveautomation.ignition.common.licensing.LicenseState;
import com.inductiveautomation.ignition.common.script.ScriptManager;
import com.inductiveautomation.ignition.common.script.hints.PropertiesFileDocProvider;
import com.inductiveautomation.ignition.gateway.clientcomm.ClientReqSession;
import com.inductiveautomation.ignition.gateway.model.AbstractGatewayModuleHook;
import com.inductiveautomation.ignition.gateway.model.GatewayContext;
import com.skm.sftp.AbstractScriptModule;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;


@SuppressWarnings("unused")
public class GatewayHook extends AbstractGatewayModuleHook {

  private  final Logger logger = LoggerFactory.getLogger(getClass());
  private final AbstractScriptModule scriptModule = new GatewayScriptModule();

  @Override
  public void setup(GatewayContext gatewayContext){logger.info("setup()");}

  @Override
  public void startup(LicenseState licenseState) {
    logger.info("startup()");
  }

  @Override
  public void shutdown(){
    logger.info("shutdown");
  }

  @Override
  public void initializeScriptManager(ScriptManager manager){
    super.initializeScriptManager(manager);
    manager.addScriptModule(
        "system.sftp",
        scriptModule,
        new PropertiesFileDocProvider());
  }

  @Override
  public Object getRPCHandler(ClientReqSession session, String projectName){
    return scriptModule;
  }
}

File: gateway/bin/main/com/skm/sftp/gateway/GatewayHook.kt
package com.skm.sftp.gateway

import com.inductiveautomation.ignition.gateway.model.AbstractGatewayModuleHook
import com.skm.sftp.AbstractScriptModule.Companion.modulePath
import com.skm.sftp.AbstractScriptModule
import com.inductiveautomation.ignition.gateway.model.GatewayContext
import com.inductiveautomation.ignition.common.licensing.LicenseState
import com.inductiveautomation.ignition.common.script.ScriptManager
import com.inductiveautomation.ignition.common.script.hints.PropertiesFileDocProvider
import com.inductiveautomation.ignition.gateway.clientcomm.ClientReqSession
import org.slf4j.LoggerFactory

@Suppress("unused")
class GatewayHook : AbstractGatewayModuleHook() {
    private val logger = LoggerFactory.getLogger(javaClass)
    private val scriptModule: AbstractScriptModule = GatewayScriptModule()

    override fun setup(gatewayContext: GatewayContext) {
        logger.info("setup()")
    }
    override fun startup(licenseState: LicenseState) {
        logger.info("startup()")
    }
    override fun shutdown() {
        logger.info("shutdown()")
    }

    override fun initializeScriptManager(manager: ScriptManager) {
        super.initializeScriptManager(manager)
        manager.addScriptModule(
            modulePath,
            scriptModule,
            PropertiesFileDocProvider()
        )
    }

    override fun getRPCHandler(
        session: ClientReqSession,
        projectName: String
    ): Any {
        return scriptModule
    }
}